spring-boot-project/spring-boot-actuator-autoconfigure/src/main/java/org/springframework/boot/actuate/autoconfigure/tracing/TracingEnvironmentPostProcessor.java

```diff
@@ -0,0 +1,78 @@
+/*
+ * Copyright 2013-2021 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.springframework.boot.actuate.autoconfigure.tracing;
+
+import java.util.HashMap;
+import java.util.Map;
+
+import org.springframework.boot.SpringApplication;
+import org.springframework.boot.env.EnvironmentPostProcessor;
+import org.springframework.core.env.ConfigurableEnvironment;
+import org.springframework.core.env.MapPropertySource;
+import org.springframework.core.env.MutablePropertySources;
+import org.springframework.core.env.PropertySource;
+
+/**
+ * Adds default properties for the application: logging pattern level that prints trace
+ * information (e.g. trace ids).
+ *
+ * @author Dave Syer
+ * @author Marcin Grzejszczak
+ * @since 3.0.0
+ */
+class TracingEnvironmentPostProcessor implements EnvironmentPostProcessor {
+
+	static final String PROPERTY_SOURCE_NAME = "defaultProperties";
```

Comment:
I'm not sure that this should mess with the `defaultProperties` property source. It may be better for it to add its own property source so that it's more clear in things like the `env` endpoint where the properties have come from.